docs/madr/decisions/008-mesh-logging.md

```diff
@@ -0,0 +1,367 @@
+# MeshLogging
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4733
+
+# Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a `MeshLogging`.
+
+# Considered Options
+
+* Create a MeshLogging policy
+  * Keep backends in the `Mesh` resource for now, move them into `MeshLoggingBackend` policy in the future
+  * Move backend to the `MeshLoggingBackend` policy now
+  * Embed backend in `MeshLogging` policy
+
+# Decision Outcome
+
+To be determined.
+
+## Overview
+
+MeshLogging allows users to log incoming and outgoing traffic for services.
+This feature is useful in the following scenarios:
+* auditing - services that contain important data might want to log access to its resources for auditing purposes
+* debugging - an engineer might want to see traffic going in/out of a service to debug misbehaviour
+* insights - logs are useful when trying to understand users and their behaviour
+
+## Specification
+
+### Naming
+
+During the weekly meeting we started talking about the appropriate name for this policy, here are the names mentioned:
+- `MeshAccessLog`
+- `MeshLog`
+- `MeshTrafficLog`
+- `TrafficLogging`
+- `Logging`
+- `MeshLogging`
+
+Please vote on your preferred name in the comments. Feel free to edit this document and add your suggestions.
+
+### Matching
+
+There are 3 parts to matching, each meaning a different part of the request flow:
+- `spec.targetRef` - where the policy attaches to, `to` and `from` are relative to this element(s)
+- `spec.to.targetRef` - outbound(s) of the `spec.targetRef`
+- `spec.from.targetRef` - inbound(s) of the `spec.targetRef`
+
+#### Top level
+
+`spec.targetRef` can have the following kinds: `Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute`
+
+Matching on `MeshGatewayRoute` and `MeshHTTPRoute` can be achieved by using a
+[HeaderMatcher](https://github.com/envoyproxy/envoy/blob/23a9a686bb4237934cd575d8e62d3e0df98b59ee/api/envoy/config/accesslog/v3/accesslog.proto#L207)
+with a value of `:path`.
+When using these targets, defining `spec.to.targetRef` is **disallowed** because
+there is no way of knowing if the outgoing request is connected to an incoming route.
+
+#### From level
+
+`spec.from.targetRef` can only have: `Mesh` for now.
+
+Matching on `MeshGatewayRoute` and `MeshHTTPRoute` does not make sense (there is no `route` that a request originates **from**).
+
+#### To level
+
+`spec.to.targetRef` can only have: `Mesh|MeshService` for now.
```

Comment:
In the future, when `spec.targetRef` is `Mesh|MeshSubset|MeshService|MeshServiceSubset` then we can have `MeshHTTPRoute` as an `Outbound` route and then we can only have `to` (`from` always goes from the application).